### PR TITLE
cleanup & preserve fzf query between ctrl+{f,g}

### PR DESCRIPTION
**Pull request by laur89**
- minor bash cleanup, such as:
      - variable localization
      - use bash syntax (eg [[]] vs [])
      - improve param quoting
      - capitalize global parameters
    - preserve user input query in fzf between ctrl+f & ctrl+g options by
      utilizing --print-query fzf option;
    - instead of recursively calling capture(), run the command in a loop
      in capture();
    - add colors to fzf header

### CODE REVIEW

**Comment by laktak** [comment]
Thanks for your PR! I appreciate your effort but you are changing a lot all at once.

What is the reason for changing all variables to upper case in the "minor bash cleanup"?

Preserving the query is a great feature but please don't change the structure (adding the loop) at the same time - it makes it hard to follow what you are doing.

Could you remove the cleanup and the loop from this PR for now?

**Comment by laur89** [comment]
PR is rather large, but features are separated by commit.

> What is the reason for changing all variables to upper case
It's a common convention to denote global variables.

> Preserving the query is a great feature but please don't change the structure (adding the loop) at the same time - it makes it hard to follow what you are doing.

Majority of the implementation remains same, it's just the indentation-shift that makes it seem like a major change.

I'll try and break changes up further.

**Comment by laur89** [comment]
Let's continue on https://github.com/laktak/extrakto/pull/49
